### ISSUE TITLE
[A11Y][Dialog]: Focus moves out of the dialog when swiping right in Safari with VoiceOver enabled

### ISSUE THREAD

**Issue by ivatrifonova** [SAP/ui5-webcomponents#11277]
### Describe the bug

According to [WCAG Technique H102](https://www.w3.org/WAI/WCAG22/Techniques/html/H102.html), focus within a modal dialog should be constrained to the content of the dialog and the browser's chrome. However, in Safari with VoiceOver enabled, the following issues are observed:

1. When swiping right, focus moves through the dialog content and the browser's navigation (chrome). However, when it reaches the last element of the navigation, focus does not return to the dialog as it should, disrupting the intended focus loop.
2.  When swiping left, focus incorrectly moves to background elements outside the dialog, violating the expected focus constraints.

This issue can also be reproduced in the Storybook implementation of the component.

### Isolated Example

https://sap.github.io/ui5-webcomponents-react/v2/iframe.html?globals=&args=&id=modals-popovers-dialog--default&viewMode=story

### Reproduction steps

1. Navigate to https://sap.github.io/ui5-webcomponents-react/v2/iframe.html?globals=&args=&id=modals-popovers-dialog--default&viewMode=story
2. Swipe right until the focus reaches the last element.

### Expected Behaviour

When using Safari with VoiceOver enabled:

1. Focus Constraint: Focus should be limited to the content of the dialog and the browser's chrome (navigation). Background elements outside the dialog should not be accessible via swiping gestures (either left or right).
2. Focus Looping: When swiping right through the dialog content and browser navigation, focus should return to the beginning of the dialog once it reaches the last focusable element, ensuring a seamless focus loop.
3. Swiping Left: When swiping left, focus should correctly move backward within the dialog and browser chrome without accessing any background elements.

### Screenshots or Videos

https://github.com/user-attachments/assets/6c41905c-1583-47ae-b050-2016672bcfc3

https://github.com/user-attachments/assets/a1c65af9-d602-41b6-a505-cd4692c95fd4

### UI5 Web Components for React Version

2.8.1

### UI5 Web Components Version

2.8.0

### Browser

Safari

### Operating System

iOS

### Relevant log output

```shell

```

### Declaration

- [x] I’m not disclosing any internal or sensitive information.

**Comment by plamenivanov91** [comment]
Hello @SAP/ui5-webcomponents-topic-rd ,

The reporter would like to constrain focus from moving out of the dialog and have focus looping when we are on iOS and using voice over.

I tried the reproduction steps on my iPhone and in both chrome and safari the "open dialog" button becomes focused when swiping left while we are on the first element in the dialog.

Could you please see if we can fix this?

Regards,
Plamen Ivanov

**Comment by dimovpetar** [comment]
Same issue is reproducible here https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/dialog/ on Android devices with TalkBack.

**Comment by GerganaKremenska** [comment]
Hello @ivatrifonova,

UI5 Web Components provide the prerequisites for screen reader support based on the HTML, ARIA, and WCAG standards. All screen readers that follow those standards should work fine. Nevertheless, there are deviations in the interpretation depending on the combination of browser and screen reader. UI5 Web Components focus on compliance with the standards, rely on automated checks for accessibility and manual tests with reference testing environments. 

When reporting issues with different testing environments, please ensure that the issue is not false positive, a real accessibility concern is present and there is an impact over the users. Therefore, we recommend to retest the issue using the mentioned reference testing tools and additionally with plain HTML.

As we have retested with plane html from the [W3 patterns](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/dialog/ ) and the behavior is presistent, we are closing the issue.

Best Regards,
Gergana
